rpc/src/rpc.rs

```diff
@@ -1939,17 +1939,23 @@ impl JsonRpcRequestProcessor {
                     amount: Reverse(amount_u64),
                 };
 
-                // Add if balance is larger than smallest item in heap.
-                // Reverse requires < to check for larger, e.g. Reverse(5) < Reverse(3)
+                // Auxiliary booleans for evict_one & add_to_heap conditions
                 let is_not_full = token_accounts_min_heap.len() < NUM_LARGEST_ACCOUNTS;
+                // Reverse requires < to check for larger, e.g. Reverse(5) < Reverse(3).
+                // Note that Some(...) < None evaluates to false, but add_to_heap will be true
+                // because None is only returned when the heap is empty (is_not_full = true)
                 let smallest_in_heap = token_accounts_min_heap.peek();
                 let is_larger = Some(&wrapped_token_account_balance) < smallest_in_heap;
-                if is_not_full | is_larger {
```

Comment:
read the logic here wrong initially 

the reason for the change is that this would evict from the heap whenever the heap was full, even in cases where `is_larger == false`